### DISCUSSION TITLE
How to create custom attribute for columns

### DISCUSSION THREAD

**Discussion by DrastiGohel** [serenity-is/Serenity#6597] [Help]
Hello all,

I am trying to create custom attribute for columns.

**Steps followed are below**: 
1. Create Attribute .ts file:
![image](https://user-images.githubusercontent.com/98172041/204582577-77c2bdea-2797-4100-a753-c69c29a31f60.png)
2. . Rebuild/transform templates - this does not generate anything in Import/ClientType
3.  Add that to column - When adding it to column it says, that file does not exist

**Trying to replicate Order table Freight Column here** - also want to replicate other columns with icons such as Employee and Ship Via
![image](https://user-images.githubusercontent.com/98172041/204589481-6611525e-9997-4234-9b2e-cf89fe5df040.png)

This is for Serenity **6.3.2**

**Comment by volkanceylan** [comment]
First, you miss a registration name inside registerFormatter call. And probably dont have the latest sergen installed. E.g. `dotnet tool update sergen`. Please read recent release notes (https://serenity.is/docs/release-notes/6.3.0) and blog posts about ES modules.

**Comment by DrastiGohel** [comment]
Hi Volkan,

Thank you for response.

I updated my sergen to latest 6.3.6.

The registration name is : @Decorators.registerFormatter('MyProject.Common.FreightFormatter')

FreightFormatterAttribute.cs is generated now

But still I cannot use that in my column.cs 

**Note**: Formatter is in Common folder and Column.cs is in Default folder
Also, the Import Folder does not seems to be included in project:
![image](https://user-images.githubusercontent.com/98172041/205353976-aaf62c0f-acd8-4e75-b686-69558fa6f368.png)


Here is the error:
![image](https://user-images.githubusercontent.com/98172041/205353707-7b04ea48-680e-4e03-863e-ecc959cb7bc1.png)

**Reply by VictorTomaili** [reply]
While using sergen to generate types, these folders should be included into your project. In StartSharp, you may have Pro.Coder package instead. In this case, the csharp code is generated by source generator on the fly.
